src/utils/intersection.js

```diff
@@ -86,9 +86,20 @@ export function measureIntersection(el) {
 export function intersectionEntryToJson(entry) {
   return {
     time: entry.time,
-    rootBounds: layoutRectFromDomRect(entry.rootBounds),
-    boundingClientRect: layoutRectFromDomRect(entry.boundingClientRect),
-    intersectionRect: layoutRectFromDomRect(entry.intersectionRect),
+    rootBounds: safeLayoutRectFromDomRect(entry.rootBounds),
+    boundingClientRect: safeLayoutRectFromDomRect(entry.boundingClientRect),
+    intersectionRect: safeLayoutRectFromDomRect(entry.intersectionRect),
     intersectionRatio: entry.intersectionRatio,
```

Comment:
haven't looked up the spec but just wanted to ask for my own knowledge if intersectionRatio here is guaranteed vs the other fields

Reply:
I know there is a special casing of cross-document tracking where many of the fields end up not filled in. I'll double check which ones.

The reason it is safe for `time` and `intersectionRatio` in particular to drop in is because they are only numbers (or potentially undefined) and so at least wouldn't throw if not present.

Not for this PR: but may want to also handle `undefined` in `safeLayoutRectFromDomRect`